modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableHeaderRow.java

```diff
@@ -314,9 +329,15 @@ final BooleanProperty reorderingProperty() {
     private final ReadOnlyObjectProperty<NestedTableColumnHeader> rootHeaderProperty() {
         return rootHeader.getReadOnlyProperty();
     }
-    final NestedTableColumnHeader getRootHeader() {
+
+    /**
+     * @return The root header that is actually just one NestedTableColumnHeader that spans
+     * the entire width.
+     */
+    public final NestedTableColumnHeader getRootHeader() {
```

Comment:
I initially made this protected but I'm using it elsewhere so I believe it should be made public.

Reply:
There are several methods gravitating around the TableView headers. 

One should understand that the TableView header is composed of one TableHeaderRow, that contains a single NestedTableColumnHeader that spans the whole width, and then several NestedTableColumnHeader that will represent each column.

Anyone who truly wants to tweak the TableView header will have to access to any methods that are linking any of these objects together. That includes getTableHeaderRow() that allow a custom TableColumnHeader to retrieve the TableHeaderRow is is contained in.

It also includes the getRootHeader() method that allow a TableColumnHeader to retrieve the root NestedTableCOlumnHeader containing all tableColumnHeader.
For example, getRootHeade() allow to add specific listener when the header is clicked to trigger a selection : https://bitbucket.org/controlsfx/controlsfx/src/1334a58d91017c1a1c4c13d9d2bb780d2fa3223c/controlsfx/src/main/java/impl/org/controlsfx/spreadsheet/HorizontalHeader.java?at=default&fileviewer=file-view-default#HorizontalHeader.java-185